### PR TITLE
Feature/deeptoolsintervals

### PR DESCRIPTION
**Pull request by dpryan79**
This implements #71 and #76, which results in almost entirely removing bx-python. Given the amount of change involved, it'd be good if we use this for a while internally and then add a "deepTools 2.3 beta" to our public Galaxy instance before making a real release.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -28,9 +28,10 @@ install:
 
 # command to run tests
 script:
-  - flake8 . --exclude=.venv,.build,foo --ignore=E501,F403,E402
-  - nosetests --with-doctest --ignore-files=plotCoverage deeptools
-  - nosetests deeptools/test/test_plotCoverage.py
+  - flake8 . --exclude=.venv,.build,foo,build,deeptoolsintervals/__init__.py --ignore=E501,F403,E402
```
Comment (bgruening) [inline .travis.yml]: foo, build?

File: deeptools/bamCompare.py [hunk 2]
```diff
@@ -186,13 +190,18 @@ def get_scale_factors(args):
                     (scalefactors_dict['sites_sampled'],
                      args.sampleLength)
 
-                print "size factor if the number of mapped " \
+                print "ignoring filtering, size factors if the number of mapped " \
```
Comment (bgruening) [inline deeptools/bamCompare.py]: At some point we should get rid of these print statements.

File: deeptools/bamCompare.py [hunk 3]
```diff
@@ -186,13 +190,18 @@ def get_scale_factors(args):
...
                 print tuple(
                     float(min(bam1.mapped, bam2.mapped)) / np.array([bam1.mapped, bam2.mapped]))
 
         elif args.scaleFactorsMethod == 'readCount':
-            scale_factors = float(min(bam1_mapped, bam2_mapped)) / np.array([bam1_mapped, bam2_mapped])
+            args.bam = args.bamfile1
+            args.scaleFactor = 1.0
+            s1 = get_scale_factor(args)
+            args.bam = args.bamfile2
+            s2 = get_scale_factor(args)
```
Comment (bgruening) [inline deeptools/bamCompare.py]: I guess a cleaner solution would be to give the bam file as second parameter to get_scale_factor()

File: .travis.yml (hunk 1 above)
Comment (dpryan79) [inline .travis.yml]: `build/` is created with `python ./setup.py install`. `foo/` is a venv for the planemo tests.

File: deeptools/bigwigCompare.py [hunk 4]
```diff
@@ -51,17 +51,19 @@ def parse_arguments(args=None):
                         required=False)
 
     parser.add_argument('--ratio',
```
Comment (bgruening) [inline deeptools/bigwigCompare.py]: This is a shared one, with e.g. bamCompare.

File: deeptools/countReadsPerBin.py [hunk 5]
```diff
@@ -269,6 +270,18 @@ def run(self):
             # in case a region is used, append the tilesize
             self.region += ":{}".format(self.binLength)
 
+        # Handle GTF options
+        transcriptID = "transcript"
+        exonID = "exon"
+        transcript_id_designator = "transcript_id"
```
Comment (bgruening) [inline deeptools/countReadsPerBin.py]: Can we put this block into a private function as utils, this at least two times.

Comment (dpryan79) [inline deeptools/countReadsPerBin.py]: I suppose so.

**Comment by dpryan79** [comment]
Closing due to accidentally targeting to `main`. Anyway, I'll just `git flow feature finish` this.

**Pull request closed without merging**